driver/makedepend.ml

```diff
@@ -277,19 +277,17 @@ let rec lexical_approximation lexbuf =
 
 let read_and_approximate inputfile =
   Depend.free_structure_names := String.Set.empty;
-  let ic = open_in_bin inputfile in
+  In_channel.with_open_bin inputfile @@ begin fun ic ->
   try
     seek_in ic 0;
     Location.input_name := inputfile;
     let lexbuf = Lexing.from_channel ic in
     Location.init lexbuf inputfile;
-    lexical_approximation lexbuf;
-    close_in ic;
-    !Depend.free_structure_names
+    lexical_approximation lexbuf
   with exn ->
-    close_in ic;
-    report_err exn;
-    !Depend.free_structure_names
+    report_err exn
+  end;
```

Comment:
I have the impression than it might be clearer to move the `with_open_bin` under the `try` branch since it will reraise the exception (and this gives us the same clean-up order).